### PR TITLE
[SignalR] Add note on Sticky Sessions

### PR DESCRIPTION
**Pull request by anurse**
Fixes https://github.com/aspnet/AspNetCore/issues/13944

Added a note on the Sticky Session requirement in the hosting overview.

[Internal Review Link](https://review.docs.microsoft.com/en-us/aspnet/core/signalr/scale?view=aspnetcore-2.1&branch=pr-en-us-14392)

### CODE REVIEW

File: aspnetcore/signalr/scale.md [hunk 1]
```diff
@@ -15,6 +15,18 @@ By [Andrew Stanton-Nurse](https://twitter.com/anurse), [Brady Gaster](https://tw
 
 This article explains hosting and scaling considerations for high-traffic apps that use ASP.NET Core SignalR.
 
+## Sticky Sessions
+
+SignalR requires that all HTTP requests for a specific connection be handled by the same server process. This means that "sticky sessions" (also called "session affinity" by some load balancers or "ARR affinity" in Azure App Service) are required in order to use SignalR. The **only** circumstances in which sticky sessions are **not** required are:
+
+1. When hosting on a single server, in a single process.
+1. When using the Azure SignalR Service.
+1. When all clients are configured to **only** use WebSockets, and the [SkipNegotiation setting](xref:signalr/configuration#configure-additional-options) is enabled in the client configuration.
```
Comment (BrennanConroy) [inline aspnetcore/signalr/scale.md]:
```suggestion
1. When all clients are configured to **only** use WebSockets, **and** the [SkipNegotiation setting](xref:signalr/configuration#configure-additional-options) is enabled in the client configuration.
```

File: aspnetcore/signalr/scale.md [hunk 2]
```diff
@@ -15,6 +15,18 @@ By [Andrew Stanton-Nurse](https://twitter.com/anurse), [Brady Gaster](https://tw
...
+
+SignalR requires that all HTTP requests for a specific connection be handled by the same server process. This means that "sticky sessions" (also called "session affinity" by some load balancers or "ARR affinity" in Azure App Service) are required in order to use SignalR. The **only** circumstances in which sticky sessions are **not** required are:
+
+1. When hosting on a single server, in a single process.
+1. When using the Azure SignalR Service.
+1. When all clients are configured to **only** use WebSockets, **and** the [SkipNegotiation setting](xref:signalr/configuration#configure-additional-options) is enabled in the client configuration.
+
+In all other circumstances (including when the Redis backplane is used), the server environment must be configured for sticky sessions.
+
+See [our documentation on publising to Azure App Service](xref:signalr/publish-to-azure-web-app) for guidance on configuring Azure App Service for SignalR.
```
Comment (scottaddie) [inline aspnetcore/signalr/scale.md]:
```suggestion
For guidance on configuring Azure App Service for SignalR, see <xref:signalr/publish-to-azure-web-app>.
```

File: aspnetcore/signalr/scale.md [hunk 3]
```diff
@@ -15,6 +15,18 @@ By [Andrew Stanton-Nurse](https://twitter.com/anurse), [Brady Gaster](https://tw
 
 This article explains hosting and scaling considerations for high-traffic apps that use ASP.NET Core SignalR.
 
+## Sticky Sessions
+
+SignalR requires that all HTTP requests for a specific connection be handled by the same server process. This means that "sticky sessions" (also called "session affinity" by some load balancers or "ARR affinity" in Azure App Service) are required in order to use SignalR. The **only** circumstances in which sticky sessions are **not** required are:
```
Comment (Rick-Anderson) [inline aspnetcore/signalr/scale.md]: SignalR requires that all HTTP requests for a specific connection be handled by the same server process. When SignalR is running on a server farm (multiple servers), "sticky sessions" must be used. "Sticky sessions" are also called session affinity by some load balancers. Azure App Service use [Application Request Routing](https://www.iis.net/learn/extensions/planning-for-arr) (ARR) to route all subsequent requests to the same server. The **only** circumstances in which sticky sessions are **not** required are:

Comment (Rick-Anderson) [reply aspnetcore/signalr/scale.md]: For some reason it's not offering to commit this suggestion. Let me know if you'd like me to do that and S&M (squash and merge).

Comment (anurse) [reply aspnetcore/signalr/scale.md]: That's still not quite right. ARR **doesn't** provide affinity by default, it *has to be enabled*. How about something like the below content. If you're good with it, go ahead and apply it and squash and merge.

A second option would be to just link to our existing doc content on configuring Azure App Service (https://docs.microsoft.com/en-us/aspnet/core/signalr/publish-to-azure-web-app?view=aspnetcore-2.2#configure-the-app-in-azure-app-service). I do that in the final sentence of this change but we could move that up here. That may be better than trying to explain all of this in a single paragraph (without run-on sentences or overloading the paragraph)

---

SignalR requires that all HTTP requests for a specific connection be handled by the same server process. When SignalR is running on a server farm (multiple servers), "sticky sessions" must be used. "Sticky sessions" are also called session affinity by some load balancers. Azure App Service uses [Application Request Routing](https://docs.microsoft.com/iis/extensions/planning-for-arr/application-request-routing-version-2-overview) (ARR) to route requests. Enabling the "ARR Affinity" setting in your Azure App Service will enable "sticky sessions". The only circumstances in which sticky sessions are not required are:

**Pull request merged**